tests/ui/traits/resolve-impl-before-constrain-check.rs

```diff
@@ -1,12 +1,15 @@
 // Need a different module so we try to build the mir for `test`
 // before analyzing `mod foo`.
+#![feature(sized_hierarchy)]
 
 mod foo {
+    use std::marker::PointeeSized;
+
     pub trait Callable {
         fn call();
     }
 
-    impl<V: ?Sized> Callable for () {
+    impl<V: PointeeSized> Callable for () {
     //~^ ERROR the type parameter `V` is not constrained by the impl trait, self type, or predicates
         fn call() {}
     }
```

Comment:
This error still fires for `impl<V: ?Sized>` right? They probably don't need to use `feature(sized_hierarchy)`?

Reply:
It does but it produces a second error, so I added the feature so that what the test was doing remained the same. If we'd prefer to leave it unmodified and test something slightly different (`V: const MetaSized`) then I can do that too.